### PR TITLE
Add the led_fade demo which ships on HiFive1 boards

### PR DESCRIPTION
**Pull request by mwachs5**
This is the example program which ships on the HiFive1 boards.

### CODE REVIEW

File: software/led_fade/led_fade.c [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+  PWM1_REG(PWM_CMP0)  = 0;
+
+  GPIO_REG(GPIO_IOF_SEL)    |= ( (1 << GREEN_LED_OFFSET) | (1 << BLUE_LED_OFFSET) | (1 << RED_LED_OFFSET));
+  GPIO_REG(GPIO_IOF_EN )    |= ( (1 << GREEN_LED_OFFSET) | (1 << BLUE_LED_OFFSET) | (1 << RED_LED_OFFSET));
+  GPIO_REG(GPIO_OUTPUT_XOR) &= ~( (1 << GREEN_LED_OFFSET) | (1 << BLUE_LED_OFFSET));
+  GPIO_REG(GPIO_OUTPUT_XOR) |= (1 << RED_LED_OFFSET);
+
+  while(1){
+    volatile uint64_t *  now = (volatile uint64_t*)(CLINT_BASE_ADDR + CLINT_MTIME);
+    volatile uint64_t then = *now + 100;
```
Comment (dns2utf8) [inline software/led_fade/led_fade.c]: Why is then volatile?

Comment (mwachs5) [inline software/led_fade/led_fade.c]: no good reason. Probably a vestige of an earlier version of the code.

**Comment by dns2utf8** [comment]
Would you mind if I merged some of my code into yours? Since it does basically the same it may be preferable to keep just one.

**Comment by mwachs5** [comment]
I think it is best to keep just one. I think people should have the right to know what is actually shipped on their board, so I would prefer to keep the led_fade demo as-as.

**Comment by dns2utf8** [comment]
While I agree with knowing what is inside the box, I think this code will be copied many times because it is **the** example inside the sdk for working with the LED.

A compromise would be to link to the current commit from the readme and cleanup the example.
What do you think?

**Pull request merged**